src/main/java/duke/service/Service.java

```diff
@@ -0,0 +1,35 @@
+package duke.service;
+
+import duke.data.IStorage;
+import duke.entities.Task;
+import duke.entities.Tasklist;
+
+public class Service {
+    public Tasklist list;
+    private IStorage storage;
+    public Ui ui;
+
+    public Service(IStorage storage, Ui ui) {
+        this.storage = storage;
+        this.list = storage.load();
+        this.ui = ui;
+    }
+
+    public void updateStorage() {
+        this.storage.write(this.list);
+    }
+
+    public void addToList(Task t) {
+        this.list.add(t);
+        int size;
+        if ((size = this.list.size()) == 1) {

```

Comment:
```suggestion
        int size = this.list.size();
        if (size == 1) {

```
Not a coding violation here, just wondering if this logic could be simplified